Dockerfile

```diff
@@ -0,0 +1,47 @@
+ARG GO_VERSION=1.18
+ARG RUNC_VERSION=v1.1.3
+
+# syntax = docker/dockerfile:1.4.0
+FROM --platform=${BUILDPLATFORM} golang:${GO_VERSION}-alpine AS base
+WORKDIR /src
+ENV CGO_ENABLED=0
+COPY go.* .
+# https://go.dev/ref/mod#module-cache
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
```

Comment:
`base` is only used by stage `build`. The stage `build` looks like it caches the go modules at `/go/pkg/mod`. So this pre-filling of modules seems to be unneeded.

Reply:
> So this pre-filling of modules seems to be unneeded.

Sorry, don't understand what you mean here, can you give a bit more context?

Reply:
Downloading go modules will be done in the stage `build` and the modules are also cached because `--mount=type=cache,target=/go/pkg/mod` is specified in that stage. So this line running `go mod download` and cacheing go modules looks duplicated with the stage `build`. I thought we can safely eliminated this. WDYT?

Reply:
Ah okay, now I get it, yes you are right, removed.